### PR TITLE
Add -r flag for matching Git branches with a regexp

### PR DESCRIPTION
**Pull request by brianrodri**
I've found it to be very helpful to be able to match a regexp rather than an exact GitHub branch. In my use case, I do this to choose between a Python 2 layout vs Python 3, for example (included in documentation):

```bash
if on_git_branch -r '.*py2'; then
  layout python2
else
  layout python
fi
```

### CODE REVIEW

File: stdlib.sh [hunk 1]
```diff
@@ -1202,9 +1211,14 @@ on_git_branch() {
     return 1
   elif [ -z "$1" ]; then
     return 0
+  elif [ "$1" = "-r" ] && [ -z "$2" ]; then
```
Comment (zimbatm) [inline stdlib.sh:1214]:
```suggestion
  elif [[ "$1" = "-r" && -z "$2" ]]; then
```
Let's use bash tests everywhere.

File: stdlib.sh [hunk 2]
```diff
@@ -1202,9 +1211,14 @@ on_git_branch() {
...
+    log_error "missing regexp pattern after \`-r\` flag"
+    return 1
   fi
   watch_file "$git_dir/HEAD"
-  [ "$(git branch --show-current)" = "$1" ]
+  local git_branch
+  git_branch=$(git branch --show-current)
+  [ "$1" = "$git_branch" ] || [ "$1" = '-r' ] && [[ "$git_branch" =~ $2 ]]
```
Comment (zimbatm) [inline stdlib.sh:1221]: The logic would work unless `"$1" = "-r"`. This is unlikely but it's better to use `if [[ "$1 = '-r' ]]; then [[ $git_branch" =~ $2 ]] else [[ "$1" = "$git_branch" ]]; fi`

**Review by zimbatm — COMMENTED** [review]
sounds good!

File: stdlib.sh (hunk 1 above)
Comment (brianrodri) [reply stdlib.sh:1214]: Done!

**Comment by zimbatm** [comment]
thanks!

**Pull request merged**